src/wallet/crypter.cpp

```diff
@@ -261,6 +287,67 @@ bool CCryptoKeyStore::Unlock(const CKeyingMaterial& vMasterKeyIn)
     return true;
 }
 
+bool CCryptoKeyStore::SetHDSeed(const HDSeed& seed)
+{
+    {
+        LOCK(cs_SpendingKeyStore);
+        if (!IsCrypted()) {
+            return CBasicKeyStore::SetHDSeed(seed);
+        }
+
+        if (IsLocked())
```

Comment:
On the surface calling `IsLocked()` after having called Lock(...) seems counterintuitive. Drilling into the `IsLocked()` method, it is still a bit confusing as much of what it does seems to overlap with what we do above.

why not say: 
```
LOCK(cs_SpendingKeyStore);
if (IsCrypted() && vMasterKey.empty()) {
	return false;
} else if (!IsCrypted()) {
	return CBasicKeyStore::SetHDSeed(seed);
}

std::vector<unsigned char> vchCryptedSecret;
// ...
```
I am not sure if this is less confusing.

Reply:
These are two semantically-different locking operations. The `LOCK` call is locking the mutex that protects the shielded keys. `IsLocked()` is checking whether the encrypted wallet is locked (and thus can't encrypt or decrypt anything) or not. It happens to lock on the mutex that protects the transparent keys; we probably didn't need to use separate mutexes, but that's an unrelated issue.